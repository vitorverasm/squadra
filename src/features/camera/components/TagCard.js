import React from 'react';
import {
  StyleSheet, Text, TouchableOpacity, View
} from 'react-native';
import { human, sanFranciscoWeights } from 'react-native-typography';
import Icon from 'react-native-vector-icons/Feather';
import { colors } from '../../../utils/styles';

const TagCard = ({ tagName, checked, onPress }) => (
  <View style={[styles.container, { backgroundColor: checked ? colors.primary : colors.white }]}>
    <TouchableOpacity onPress={onPress} style={styles.touchable}>
      <Icon
        name={checked ? 'check' : 'tag'}
        size={14}
        color={checked ? colors.white : colors.primary}
        style={{ backgroundColor: 'transparent' }}
      />
      <Text style={[styles.tagLabel, { color: checked ? colors.white : colors.primary }]}>
        {tagName}
      </Text>
    </TouchableOpacity>
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 0,
    flexDirection: 'row',
    borderRadius: 4,
    paddingVertical: 5,
    paddingHorizontal: 10,
    height: 35,
    marginHorizontal: 5,
    alignItems: 'center',
    justifyContent: 'center'
  },
  touchable: {
    flex: 1,
    flexDirection: 'row',
    backgroundColor: 'transparent',
    alignItems: 'center',
    justifyContent: 'center'
  },
  tagLabel: {
    ...human.footnote,
    ...sanFranciscoWeights.light,
    marginLeft: 2
  }
});

export default TagCard;
