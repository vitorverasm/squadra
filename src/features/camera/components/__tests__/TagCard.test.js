import { shallow } from 'enzyme';
import React from 'react';
import TagCard from '../TagCard';

describe('TagCard', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<TagCard tagName="teste" checked onPress={jest.fn} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
