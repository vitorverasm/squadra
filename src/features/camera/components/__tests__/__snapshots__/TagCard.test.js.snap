// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagCard renders correctly 1`] = `
"<View style={{...}}>
  <TouchableOpacity onPress={[Function: bound fn]} style={{...}} activeOpacity={0.2}>
    <Icon name=\\"check\\" size={14} color=\\"#FFFFFF\\" style={{...}} allowFontScaling={false} />
    <Text style={{...}}>
      teste
    </Text>
  </TouchableOpacity>
</View>"
`;
